test/include.test.js

```diff
@@ -60,6 +60,42 @@ describe('include', function() {
     });
   });
 
+  it('should not have changed the __strict flag of the model', function(done) {
+    const originalStrict = User.definition.settings.strict;
+    User.definition.settings.strict = true; // Change to test regression for issue #1252
+    const finish = (err) => {
+      // Restore original user strict property
+      User.definition.settings.strict = originalStrict;
+      done(err);
+    };
+    User.find({include: 'posts'}, function(err, users) {
+      if (err) return finish(err);
+      users.forEach(user => {
+        user.should.have.property('__strict', true); // we changed it
+      });
+      finish();
+    });
+  });
+
+  it('should not save in db included models, in query returned models', function(done) {
+    const originalStrict = User.definition.settings.strict;
+    User.definition.settings.strict = true; // Change to test regression for issue #1252
+    const finish = (err) => {
+      // Restore original user strict property
+      User.definition.settings.strict = originalStrict;
+      done(err);
+    };
+    User.findOne({where: {id: 1}, include: 'posts'}, function(err, user) {
+      if (err) return finish(err);
+      user.save(function(err) { // save the returned user
+        if (err) return finish(err);
+        // should not store in db the posts
+        JSON.parse(User.dataSource.adapter.cache.User[1]).should.not.have.property('posts');
+        finish();
```

Comment:
I accessed the Memory storage directly, because using a direct find, it would adhere to the defined model properties and since `posts` is not an actual model property it would not be attached to the model beucase it would be stripped of under the build process of the ModelConstructor. However it would still be in the db storage.
This it the same case with MongoDB. Retrieved models do not have the extra properties, but if you look into the database directly you will see them saved as a whole embedded document